### PR TITLE
Verify launch mount source path before launching

### LINKED ISSUE #2817: [cli/launch] `--mount` option should check if source directories exist and abort launch if they don't
Related to #2815 

Right now, when doing `launch --mount <source>:<target>`, the instance is launched and then `<source>` and `<target>` are passed to the `mount` command. If `<source>` does not exist, the mount will fail, but the instance will already be launched. We should check ahead of time if the requested source directories for mounting exist, and abort the entire launch if they don't.

### PR DESCRIPTION
**Pull request by sharder996**
Verify client source directories before launching an instance and fail the launch if they are invalid.

Fixes #2817

### CODE REVIEW

**Comment by sharder996** [comment]
Unit test has been written! @andrei-toterman

File: tests/test_cli_client.cpp [hunk 1]
```diff
@@ -1220,6 +1220,39 @@ TEST_F(Client, launch_cmd_mount_option)
...
+        send_command({"launch", "--name", instance_name, "--mount", fmt::format("{}:{}", fake_source, fake_target)},
+                     trash_stream, err),
+        mp::ReturnCode::CommandLineError);
+    EXPECT_THAT(err.str(), HasSubstr(fmt::format("Mount source path \"{}\" does not exist", fake_source)));
+
+    EXPECT_EQ(
+        send_command({"launch", "--name", instance_name, "--mount", fmt::format("{}:{}", fake_source, fake_target)},
+                     trash_stream, err),
+        mp::ReturnCode::CommandLineError);
+    EXPECT_THAT(err.str(), HasSubstr(fmt::format("Mount source path \"{}\" does not exist", fake_source)));
```
Comment (andrei-toterman) [inline tests/test_cli_client.cpp:1253]: hmm shouldn't these return different error messages? i think you should clear `err` before each `send_command` invocation, so you don't compare against errors from previous calls.

**Review by andrei-toterman — CHANGES_REQUESTED** [review]

File: tests/test_cli_client.cpp (hunk 1 above)
Comment (sharder996) [reply tests/test_cli_client.cpp]: Sorry, copy paste error! I also added a line to clear the stream to make sure that the previous output is gone.

**Review by andrei-toterman — APPROVED** [review]
thanks, looks good!

bors merge

**Comment by andrei-toterman** [comment]
a moment of appreciation for bors working on the first try

**Pull request merged**